girder/models/model_base.py

```diff
@@ -1261,11 +1261,12 @@ def copyAccessPolicies(self, src, dest, save=False):
         return dest
 
     def filterResultsByPermission(self, cursor, user, level, limit=0, offset=0,
-                                  removeKeys=()):
+                                  removeKeys=(), flags=None):
         """
         Given a database result cursor, this generator will yield only the
         results that the user has the given level of access on, respecting the
-        limit and offset specified.
+        limit and offset specified. It can also only yield results for which the
```

Comment:
I find this language a bit confusing (specificaly the 'also only yield results' part).  Maybe

    Given a database result cursor, this generator will yield only the
    results that the user has the given level of access and specified access flags on, respecting the
    limit and offset specified.

Also, do you want to support the case where someone only searches by flags and not by level, or do you always want someone to specify a level?